src/bulk/common.ts

```diff
@@ -31,37 +32,37 @@ export const BatchType = {
 export type BatchTypeId = typeof BatchType[keyof typeof BatchType];
 
 /** @public */
-export interface InsertOneModel {
+export interface InsertOneModel<TSchema> {
```

Comment:
Hi, I've actually been looking forwards to this feature. I was wondering if this could/should instead be `TSchema extends Document = Document` like I see you made some of the ones below work? Defaulting and expecting it to match the shape of a document seems like it might be helpful in preventing some errors and w/ type-ahead/type-hinting. Could maybe have it do `extends Partial<Document>` instead or something of the sorts, if the concern is not needing everything from the document

Reply:
Hey! Thanks for jumping into the conversation, I appreciate feedback from the community on this effort, looks like I already made your suggestion, feel free to call out anything else you see 😃

Reply:
Yup, awesomesauce! Thank you for the work in maintaining this library :D